source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1844,11 +1844,6 @@ bool SwiftLanguageRuntime::GetDynamicTypeAndAddress_Protocol(
     if (!type.IsValid())
       return false;
 
-    lldb::addr_t class_metadata_ptr = payload0_sp->GetAddressOf();
-    if (class_metadata_ptr == LLDB_INVALID_ADDRESS || class_metadata_ptr == 0)
-      return false;
-    address.SetRawAddress(class_metadata_ptr);
```

Comment:
`address` is passed by-reference. Do we want to still `SetRawAddress` when we get a `class_metadata_ptr`, i.e.,

```
if (class_metadata_ptr && class_metadata_ptr != LLDB_INVALID_ADDRESS)
  address.SetRawAddress(class_metadata_ptr);
```

eliminating the early exit but not the successful case?

Reply:
Most definitely.